engine/src/main/java/com/redhat/cloud/notifications/recipients/RecipientResolver.java

```diff
@@ -40,6 +41,16 @@ public Set<User> recipientUsers(String orgId, Set<RecipientSettings> requests, S
     }
 
     private Set<User> recipientUsers(String orgId, RecipientSettings request, Set<String> subscribers, boolean isOptIn) {
+        /*
+         If the subscription type is opt-in, the user preferences should NOT be ignored and the subscribers Set is empty,
+         then we don't need to retrieve the users from RBAC/IT/BOP because we'll return an empty Set anyway.
+         */
+        // TODO Report this change in the email connector ASAP.
+        if (isOptIn && !request.isIgnoreUserPreferences() && subscribers.isEmpty()) {
```

Comment:
Considering that this won't be true for the drawer, I wonder if we should have something more for fetching/caching the user's :thinking:

Reply:
I think the only reasonable approach in the long run is to make the whole users retrieval an async process. That's something I've had in mind for a while but now I'm pretty sure we'll have to implement it soon.